crates/rmf_site_editor/src/widgets/inspector/mod.rs

```diff
@@ -350,53 +358,81 @@ impl<'w, 's> WidgetSystem<Tile> for Inspector<'w, 's> {
-                        (
-                            category.map(|x| x.label()).unwrap_or("<Unknown Type>"),
-                            site_id,
-                        )
-                    } else {
-                        ("<Unknown Type>", None)
+                for e in selection.selected.iter() {
+                    let selection = match inspect_for_query.get(*e) {
+                        Ok(inspect_for) => inspect_for.entity,
+                        Err(_) => *e,
                     };
 
-                if let Some(site_id) = site_id {
-                    ui.heading(format!("{} #{}", label, site_id.0));
-                } else {
-                    ui.heading(format!("{} (unsaved)", label));
-                }
+                    let (label, site_id) =
+                        if let Ok((category, site_id)) = params.heading.get(selection) {
+                            (
+                                category
+                                    .map(|x| x.label())
+                                    .unwrap_or("<Unknown Type>")
+                                    .to_string(),
+                                site_id,
+                            )
+                        } else {
+                            ("<Unknown Type>".to_string(), None)
+                        };
 
-                let children: Result<SmallVec<[_; 16]>, _> = params
-                    .children
-                    .get(id)
-                    .map(|children| children.iter().collect());
-                let Ok(children) = children else {
-                    return;
-                };
+                    let children: Result<SmallVec<[_; 16]>, _> = params
+                        .children
+                        .get(id)
+                        .map(|children| children.iter().collect());
 
-                panel.align(ui, |ui| {
-                    for child in children {
-                        let inspect = Inspect {
-                            selection,
-                            inspection: child,
-                            panel,
-                        };
-                        let _ = world.try_show_in(child, inspect, ui);
-                    }
-                });
+                    let Ok(children) = children else {
+                        continue;
+                    };
+
+                    selection_properties.push(InspectionProperties {
+                        selection,
+                        label,
+                        site_id: site_id.clone().copied(),
+                        children,
+                    });
+                }
+
+                for InspectionProperties {
+                    selection,
+                    label,
+                    site_id,
+                    children,
+                } in selection_properties
+                {
+                    //todo(@johntgz) Find a better way to deal with egui id conflicts
```

Comment:
Once we stop displaying all the inspection data during multi-selection, we won't have to worry about this.